pkg/controllers/update.go

```diff
@@ -330,41 +335,17 @@ func isUpdateAvailableViaWinget(updateVersion string) bool {
 	return len(matches) > 0
 }
 
-func updateViaWinget(version string) (bool, string, Error) {
-	command := fmt.Sprintf("winget upgrade --id %s --exact --disable-interactivity --version %s", wingetPackageId, strings.TrimPrefix(version, "v"))
-	utils.LogDebug(fmt.Sprintf("Executing \"%s\"", command))
-
-	startTime := time.Now()
-
-	var out bytes.Buffer
-	cmd, err := utils.RunCommandString(command, os.Environ(), nil, &out, &out, true)
-	if err != nil {
-		utils.LogDebugError(err)
-		CaptureEvent("WingetUpgradeFailed", map[string]interface{}{"durationMs": 0})
-		return false, "", Error{Message: "Unable to execute winget"}
-	}
-
-	exitCode, err := utils.WaitCommand(cmd)
-
-	strOut := out.String()
-	utils.LogDebug(strOut)
+func updateViaWinget(version string) error {
+	CaptureEvent("WingetUpgradeInitiated", nil)
 
-	executeDuration := time.Since(startTime).Milliseconds()
-
-	if err != nil || exitCode != 0 {
-		var e Error
-		if strings.Contains(strOut, "No installed package found matching input criteria.") {
-			e = Error{Message: "The Doppler CLI is not installed via winget"}
-		} else if strings.Contains(strOut, "No applicable upgrade found.") || strings.Contains(strOut, "No available upgrade found.") {
-			e = Error{Message: "You are already running the latest version available via winget"}
-		} else {
-			e = Error{Err: err, Message: "Unable to upgrade via winget"}
-		}
+	c := []string{"cmd.exe", "/C", "winget", "upgrade", "--id", wingetPackageId, "--exact", "--disable-interactivity", "--version", strings.TrimPrefix(version, "v")}
 
-		CaptureEvent("WingetUpgradeFailed", map[string]interface{}{"durationMs": executeDuration})
-		return false, "", e
+	utils.LogDebug(fmt.Sprintf("Executing \"%s\"", strings.Join(c[2:], " ")))
+	cmd := exec.Command(c[0], c[1:]...)
+	if err := cmd.Run(); err != nil {
```

Comment:
check: Do we need to put a delay before the backgrounded `winget upgrade` command in case it starts faster than we can exit?

Reply:
Hah, spoken like someone that clearly hasn't used winget.